test/testdata/dsl/opus_enum_not_generic.rb

```diff
@@ -0,0 +1,22 @@
+# typed: strict
+
+module Opus
+  class Enum
+    extend T::Sig
+
+    sig {params(x: T.nilable(String)).void}
+    def initialize(x = nil)
+    end
+
+    sig {params(blk: T.proc.void).void}
+    def self.enums(&blk); end
+  end
+end
+
+class MyEnum < Opus::Enum
+  enums do
+  X = new
+  end
+end
```

Comment:
The key thing to note here is that before this change, this test would have failed because `type_template` would have been undefined, even though it's being generated by the DSL pass. We can get rid of both this new `extend` and the old `type_template` sends from the DSL pass after pay-server is codemodded.